packages/babel-helpers/src/index.ts

```diff
@@ -72,14 +72,17 @@ function getHelperMetadata(file: File, helperName: string): HelperMetadata {
     ExportDefaultDeclaration(child) {
       const decl = child.get("declaration");
 
-      if (!decl.isFunctionDeclaration() || !decl.node.id) {
+      if (decl.isIdentifier()) {
+        exportName = decl.node.name;
+        exportPath = makePath(child);
+      } else if (decl.isFunctionDeclaration() && decl.node.id) {
+        exportName = decl.node.id.name;
+        exportPath = makePath(child);
+      } else {
         throw new Error(
           `Helpers can only export named function declarations (in ${helperName})`,
         );
       }
-
-      exportName = decl.node.id.name;
-      exportPath = makePath(child);
```

Comment:
TypeScript really, really, really hates function as ES5 "class". So I will have to write this:

```
interface ClazzImpl {}

interface ClazzConstructor {
  new (): ClazzImpl,
  prototype: ClazzImpl
}

// Have to use variable function declaration, otherwise I can't do `as any as`
const ClazzImpl = function () {} as any as ClazzConstructor;

// Now it is not a export default function declaration
export default ClazzImpl
```

Reply:
cc @nicolo-ribaudo Any suggestions about this?

Reply:
`AsyncGenerator` is only used in the `wrapAsyncGenerator` helper. We can remove the `AsyncGenerator.js` file, and just inline it there so that the function is not exported anymore. Would that help?

Reply:
@nicolo-ribaudo And `OverloadYield` also have the same issue, Is it being used elsewhere?

Reply:
OverloadYield is imported by different other helpers.

I think we can relax the `export default` requirement on helpers to allow doing this:
```ts
declare class _OverloadYield {
  v: string;
  k: string;
}

/*
 * 'kind' is an enum:
 *   0 => This yield was an await expression
 *   1 => This yield comes from yield*
 */
function _OverloadYield(this: _OverloadYield, value: string, kind: string) {
  this.v = value;
  this.k = kind;
}
export { _OverloadYield as default };
```

I can do it as part of https://github.com/babel/babel/pull/16501, but it will take a couple days. You could work on something else in the meantime.

Reply:
Sure! I will rebase the PR once #16501 is merged.

Reply:
@nicolo-ribaudo Since `export { as default }` now works, do you still want to merge `AsyncGenerator` into the `wrapAsyncGenerator` or should I just leave it as a separate file?

Reply:
Yeah lets merge it, we don't need to keep it separate and having less helpers is generally good.